Brussels | ITP-2025-1 | Preeti Tyagi | Week 2 | Form Controls

preeti-t opened the pull request:
<!--

You must title your PR like this:

Cohort | FirstName LastName | Week | Assignment Title

For example,

ITP-2025-1 | Carol Owen | Week 1 | Alarm Clock

Fill in the template below - remove any sections that don't apply.

Complete the self checklist - replace each empty box in the checklist [ ] with a [x].

Add the label "Needs Review" and you will get review.

Respond to volunteer reviews until the volunteer marks it as "Complete".

-->

## Learners, PR Template

Self checklist

- [x] I have titled my PR with Region | Cohort | FirstName LastName | Sprint | Assignment Title
- [x] My changes meet the requirements of the task
- [x] I have tested my changes
- [x] My changes follow the [style guide](https://curriculum.codeyourfuture.io/guides/reviewing/style-guide/)

## Changelist

Created the Form control page based on assignment requirements for t-shirt 
no use of CSS and JS
accessibility score is 100

## Questions

Ask any questions you have for your reviewer.

preeti-t committed:
Update index.html

created the index html based on requirements

preeti-t committed:
Update index.html

Added some spacing to increase the accessibility score to 100

preeti-t committed:
Update index.html

spacing for input fields to increase accessibility

preeti-t committed:
Update index.html

fixing spacing

preeti-t committed:
Update index.html

added comment

serotonine commented on Form-Controls/index.html:
```diff
@@ -1,27 +1,69 @@
+      </p>
+
+      <p>
+        <label for="email">What is the customer's email? <span aria-hidden="true">*</span></label><br>
```
maybe you could set the label as block (display:block) instead of adding a `<br>` ?

serotonine commented on Form-Controls/index.html:
```diff
@@ -1,27 +1,69 @@
+      </fieldset>
+
+      <p>
+        <label for="size">What size does the customer want? <span aria-hidden="true">*</span></label><br>
```
maybe you could set the label as block (display:block) instead of adding a `<br> `?

serotonine commented on Form-Controls/index.html:
```diff
@@ -1,27 +1,69 @@
+        <label for="size">What size does the customer want? <span aria-hidden="true">*</span></label><br>
+        <select id="size" name="size" required aria-required="true">
+          <option value="" disabled selected aria-hidden="true">Choose a size</option>
+          <option value="XS">XS</option>
```
I would set in lowercase the value `<option value="xs">XS</option>`

serotonine commented on Form-Controls/index.html:
```diff
@@ -1,27 +1,69 @@
+      <p>
+        <button type="submit">Confirm selection</button>
+      </p>
+
```
I guess all the blank lines between `<p>` are not necessary

talmurshidi commented:
Well done, @preeti-t 

#### What You Did Exceptionally Well
- **Outstanding structure and semantics** — clean, accessible HTML5 with well-organized sections.
- Used a clear `<main>` element with `aria-labelledby` tied to the page title, improving **screen reader navigation**.
- Excellent use of **ARIA attributes** (`aria-required`, `aria-hidden`, `autocomplete`) — this shows strong awareness of accessibility best practices.
- Every input has a **proper `<label>`** linked with a `for` attribute.
- Applied **validation attributes**: `required`, `minlength="2"`, and appropriate `type="email"`.
- Correctly implemented **fieldset + legend** for the color group.
- Restricted color selection to 3 fixed options — compliant with the requirement.
- Size selection uses a `<select>` dropdown with a placeholder option and 6 valid sizes — perfect.
- Included a **confirmation checkbox**, adding an extra layer of user accountability.
- No CSS or JavaScript included — correct as per the assignment brief.
- Clear and professional formatting, following accessibility and readability standards.

#### Very Minor Suggestions for Improvement
- The form already meets **Lighthouse Accessibility 100/100**, but consider:
  - Adding `<fieldset>` for the **checkbox and identity fields** could slightly improve semantic grouping.
  - Avoid `<br>` tags — use semantic containers like `<p>` or `<div>` consistently for layout.
- For teaching beginners, this level of ARIA markup might be **too advanced** — consider simplifying in an introductory version.

#### Example Optional Enhancement
```html
<fieldset>
  <legend>Confirm identity</legend>
  <p>
    <label for="fullName">Customer Name:</label>
    <input id="fullName" name="fullName" type="text" required minlength="2">
  </p>
  <p>
    <label for="email">Email:</label>
    <input id="email" name="email" type="email" required>
  </p>
  <p>
    <input id="confirmAccount" name="confirmAccount" type="checkbox" required>
    <label for="confirmAccount">I confirm this is my account.</label>
  </p>
</fieldset>
```

preeti-t committed:
Update index.html

Minor changes made and added some enhancement  for improvement of the code as suggested.

preeti-t commented:
@talmurshidi Thank you for the detailed review, I have made some changes as suggested by you.

talmurshidi commented:
@preeti-t thank you for the updates. Also, thank AI for the nice help.
Please try to use AI for learning, not solving assignments.
AI is a useful tool for conversation and asking questions, providing more explanations, and examples.
The work is good, so I closed the PR as done.

preeti-t commented:
Sure.. @talmurshidi thank you for genuine review, normally I try to use AI for help only. Sorry for what you have noticed, that happened because i was in hurry to finish the task. but I will do slowly and my best for future. Thanks again.
